### PR TITLE
[COST-4230] Update csv-description: add deprecation notice

### PR DESCRIPTION
**Pull request by djnakabaale**

### CODE REVIEW

File: docs/csv-description.md [hunk 1]
```diff
@@ -19,6 +19,10 @@ The Koku Metrics Operator (`koku-metrics-operator`) collects the metrics require
 * PersistentVolumeClaim (PVC) configuration: The KokuMetricsConfig CR can accept a PVC definition and the operator will create and mount the PVC. If one is not provided, a default PVC will be created.
 * Restricted network installation: this operator can function on a restricted network. In this mode, the operator stores the packaged reports for manual retrieval.
 
+## New in v3.1.0:
+* Add service-account aunthentication type.
```
Comment (maskarb) [inline docs/csv-description.md:23]:
```suggestion
* Add service-account authentication type.
```

File: docs/csv-description.md [hunk 2]
```diff
@@ -60,8 +64,8 @@ If these assumptions are not met, the operator will not deploy correctly. In the
 
 ## Configurable parameters:
   * `authentication`:
-    * `type: token` -> The authentication method for connecting to `console.redhat.com`. The default and preferred method is `token`. `basic` and `service-account` authentication methods are used when the openshift-config pull-secret does not contain a token for `console.redhat.com`.
-    * `secret_name` -> The Secret used by the operator when the authentication type is `basic` or `service-account`. This parameter is required **only if** the authentication type is `basic` or `service-account`.
+    * `type: token` -> The authentication method for connecting to `console.redhat.com`. The default and preferred method is `token`. `basic`(deprecated) and `service-account` authentication methods are used when the openshift-config pull-secret does not contain a token for `console.redhat.com`.
```
Comment (maskarb) [inline docs/csv-description.md:67]:
```suggestion
    * `type: token` -> The authentication method for connecting to `console.redhat.com`. The default and preferred method is `token`. `basic` (deprecated) and `service-account` authentication methods are used when the openshift-config pull-secret does not contain a token for `console.redhat.com`.
```

File: docs/csv-description.md [hunk 3]
```diff
@@ -60,8 +64,8 @@ If these assumptions are not met, the operator will not deploy correctly. In the
...
+    * `secret_name` -> The Secret used by the operator when the authentication type is `basic`(deprecated) or `service-account`. This parameter is required **only if** the authentication type is `basic`(deprecated) or `service-account`.
```
Comment (maskarb) [inline docs/csv-description.md:68]:
```suggestion
    * `secret_name` -> The Secret used by the operator when the authentication type is `basic` (deprecated) or `service-account`. This parameter is required **only if** the authentication type is `basic` (deprecated) or `service-account`.
```

File: docs/csv-description.md [hunk 4]
```diff
@@ -83,21 +87,21 @@ If these assumptions are not met, the operator will not deploy correctly. In the
 ## Configure the koku-metrics-operator
 **Note** There are separate instructions for configuring the `koku-metrics-operator` to run in a restricted network.
 ##### Configure authentication
-The default authentication for the operator is `token`. No further steps are required to configure token authentication. If `basic` is the preferred authentication method, a Secret must be created which holds username and password credentials:
+The default authentication for the operator is `token`. No further steps are required to configure token authentication. If `basic`(deprecated) or `service-account` is the preferred authentication method, a Secret which holds the credentials must be created:
```
Comment (maskarb) [inline docs/csv-description.md:90]:
```suggestion
The default authentication for the operator is `token`. No further steps are required to configure token authentication. If `basic` (deprecated) or `service-account` is the preferred authentication method, a Secret which holds the credentials must be created:
```

File: docs/csv-description.md [hunk 5]
```diff
@@ -83,21 +87,21 @@ If these assumptions are not met, the operator will not deploy correctly. In the
...
 1. On the left navigation pane, select `Workloads` -> `Secrets` -> select Project: `koku-metrics-operator` -> `Create` -> `Key/Value Secret`
 2. Give the Secret a name and add 2 keys (all lowercase) for the respective authentication type. The values for these keys correspond to console.redhat.com credentials:
-    * basic auth: `username` and `password`
+    * basic auth(deprecated): `username` and `password`
```
Comment (maskarb) [inline docs/csv-description.md:93]:
```suggestion
    * basic auth (deprecated): `username` and `password`
```

File: docs/csv-description.md [hunk 6]
```diff
@@ -83,21 +87,21 @@ If these assumptions are not met, the operator will not deploy correctly. In the
...
     * service-account auth: `client_id` and `client_secret` 
 
 3. Select `Create`.
 ##### Create the KokuMetricsConfig
 Configure the koku-metrics-operator by creating a `KokuMetricsConfig`.
 1. On the left navigation pane, select `Operators` -> `Installed Operators` -> `koku-metrics-operator` -> `KokuMetricsConfig` -> `Create Instance`.
-2. For `basic` or `service-account` authentication, edit the following values in the spec:
+2. For `basic`(deprecated) or `service-account` authentication, edit the following values in the spec:
```
Comment (maskarb) [inline docs/csv-description.md:100]:
```suggestion
2. For `basic` (deprecated) or `service-account` authentication, edit the following values in the spec:
```

File: docs/csv-description.md (hunk 1 above)
Comment (djnakabaale) [reply docs/csv-description.md]: nice catch 👀

**Review by maskarb — APPROVED** [review]

**Pull request merged**